src/components/mgt-person-card/mgt-person-card.ts

```diff
@@ -0,0 +1,415 @@
+
+  /**
+   * Synchronizes property values when attributes change.
+   *
+   * @param {*} name
+   * @param {*} oldValue
+   * @param {*} newValue
+   * @memberof MgtPersonCard
+   */
+  public attributeChangedCallback(name, oldValue, newValue) {
+    super.attributeChangedCallback(name, oldValue, newValue);
+
+    // TODO: handle when an attribute changes.
+    //
+    // Ex: load data when the name attribute changes
+    // if (name === 'person-id' && oldValue !== newValue){
+    //  this.loadData();
+    // }
+
+    if (name === 'showadditionaldetails' && oldValue !== newValue) {
+      this.showadditionaldetails = 'false';
+    }
+  }
+
+  /**
+   * Invoked when the element is first updated. Implement to perform one time
+   * work on the element after update.
+   *
+   * Setting properties inside this method will trigger the element to update
+   * again after this update cycle completes.
+   *
+   * * @param _changedProperties Map of changed properties with old values
+   */
+  public firstUpdated() {
+    Providers.onProviderUpdated(() => this.loadData());
+    this.loadData();
+  }
+
+  /**
+   * Invoked on each update to perform rendering tasks. This method must return
+   * a lit-html TemplateResult. Setting properties inside this method will *not*
+   * trigger the element to update.
+   */
+  protected render() {
+    if (this.personDetails) {
+      const user = this.personDetails;
+
+      // tslint:disable-next-line: one-variable-per-declaration
+      let department, jobTitle;
+
+      if (user.department) {
+        department = html`
+          <div class="department">${user.department}</div>
+        `;
+      }
+
+      if (user.jobTitle) {
+        jobTitle = html`
+          <div class="job-title">${user.jobTitle}</div>
+        `;
+      }
+      return html`
+        <div class="root">
+          <div class="default-view">
+            ${this.renderTemplate('default', { person: this.personDetails }) ||
+              html`
+                <mgt-person .personDetails=${this.personDetails} .personImage=${this.personImage}></mgt-person>
+                <div class="details">
+                  <div class="display-name">${user.displayName}</div>
+                  ${jobTitle} ${department}
+                  <div class="base-icons">
+                    ${this.renderIcons()}
+                  </div>
+                </div>
+              `}
+          </div>
+          <div class="additional-details-container" @click=${this._showAdditionalDetails}>
+            ${this.renderAdditionalDetails()}
+          </div>
+        </div>
+      `;
+    }
+  }
+
+  private renderIcons() {
+    const user = this.personDetails;
+    // tslint:disable-next-line: one-variable-per-declaration
+    let chat, email, phone;
+    if ((user as MicrosoftGraph.User).mailNickname) {
+      chat = html`
+        <svg xmlns="http://www.w3.org/2000/svg" width="21" height="22" viewBox="0 0 21 22" fill="none">
```

Comment:
Can you move these icons in a separate file? we might need to re-use them and they make the code harder to read. You are also defining them multiple times as far as I can tell

Reply:
refactor: 
6057c3b